youtube_dl/extractor/platzi.py

```diff
@@ -104,50 +104,40 @@ def _real_extract(self, url):
-                # strict regex first
-                (r'client_data\s*=\s*({.+?})\s*;\s*\n',
-                 r'client_data\s*=\s*({.+?})\s*;'),
-                webpage, 'client data'),
+        data_preloaded_state = self._parse_json(
+            self._search_regex((r'window.__PRELOADED_STATE__ = (.*)\<\/script'), webpage, 'client data'),
             lecture_id)
 
-        material = data['initialState']['material']
-        desc = material['description']
-        title = desc['title']
-
+        # desc = data_preloaded_state['videoPlayer']['courseDescription']
+        title = data_preloaded_state['videoPlayer']['name']
+        duration = data_preloaded_state['videoPlayer']['duration']
+        servers = data_preloaded_state['videoPlayer']['video']['servers']
         formats = []
-        for server_id, server in material['videos'].items():
-            if not isinstance(server, dict):
-                continue
+        for server in servers.keys():
             for format_id in ('hls', 'dash'):
-                format_url = url_or_none(server.get(format_id))
-                if not format_url:
-                    continue
-                if format_id == 'hls':
+                server_json = servers[server]
+                if 'hls' in server_json.keys():
                     formats.extend(self._extract_m3u8_formats(
-                        format_url, lecture_id, 'mp4',
+                        server_json['hls'], lecture_id, 'mp4',
                         entry_protocol='m3u8_native', m3u8_id=format_id,
-                        note='Downloading %s m3u8 information' % server_id,
+                        note='Downloading %s m3u8 information' % server_json['id'],
                         fatal=False))
-                elif format_id == 'dash':
+                elif 'dash' in server_json.keys():
                     formats.extend(self._extract_mpd_formats(
-                        format_url, lecture_id, mpd_id=format_id,
-                        note='Downloading %s MPD manifest' % server_id,
+                        server_json['dash'], lecture_id, mpd_id=format_id,
+                        note='Downloading %s MPD manifest' % server_json['id'],
                         fatal=False))
         self._sort_formats(formats)
 
-        content = str_or_none(desc.get('content'))
-        description = (clean_html(compat_b64decode(content).decode('utf-8'))
-                       if content else None)
-        duration = int_or_none(material.get('duration'), invscale=60)
+        # content = str_or_none(data['videoPlayer']['content'])
+        # description = (clean_html(compat_b64decode(content).decode('utf-8'))
+        #               if content else None)
+        duration = int_or_none(duration, invscale=60)
 
         return {
             'id': lecture_id,
             'title': title,
-            'description': description,
+            'description': '',
```

Comment:
If there really is no `description`, omit it. The public pages have a description in `<meta name="description" content="...">` as well as in the `og:description` and `twitter:description` `<meta>` items.